lib/Surface/USB/Shuttle.js

```diff
@@ -0,0 +1,213 @@
+		this.ipcWrapper = ipcWrapper
+		this.shuttle = shuttle
+		this.deviceInfo = deviceInfo
+		this.modelInfo = modelInfo
+
+		this.config = {
+			// No config currently present
+		}
+
+		this.ipcWrapper.log('debug', `Adding Contour Shuttle USB device ${devicepath}`)
+
+		this.info = {
+			type: `Shuttle ${this.deviceInfo.name}`,
+			devicepath: devicepath,
+			configFields: [],
+			keysPerRow: this.modelInfo.totalCols,
+			keysTotal: this.modelInfo.totalCols * this.modelInfo.totalRows,
+			deviceId: `shuttle:${this.deviceInfo.id}`,
+		}
+
+		this.shuttle.on('error', (error) => {
+			console.error(error)
+			this.ipcWrapper.remove()
+		})
+
+		this.shuttle.on('buttondown', (info) => {
+			const keyIndex = buttonToIndex(this.modelInfo, info) ?? rotaryToButtonIndex(this.modelInfo, info)
+			if (keyIndex === undefined) {
+				return
+			}
+
+			this.ipcWrapper.click(keyIndex, true)
+		})
+
+		this.shuttle.on('buttonup', (info) => {
+			const keyIndex = buttonToIndex(this.modelInfo, info) ?? rotaryToButtonIndex(this.modelInfo, info)
+			if (keyIndex === undefined) {
+				return
+			}
+
+			this.ipcWrapper.click(keyIndex, false)
+		})
+
+		this.shuttle.on('jog-dir', (delta) => {
+			const keyIndex = rotaryToButtonIndex(this.modelInfo, encoders.JOG)
+			if (keyIndex === undefined) {
+				return
+			}
+
+			this.ipcWrapper.rotate(keyIndex, delta == 1)
+		})
+
+		this.shuttle.on('shuttle-trans', (previous, current) => {
+			const keyIndex = rotaryToButtonIndex(this.modelInfo, encoders.SHUTTLE)
+			if (keyIndex === undefined) {
+				return
+			}
+
+			this.ipcWrapper.rotate(keyIndex, previous < current)
+		})
+
+		this.shuttle.on('disconnect', (error) => {
+			console.error(error)
+			this.ipcWrapper.remove()
+		})
+	}
+
+	async #init() {
+		this.ipcWrapper.log('debug', `Contour Shuttle ${this.deviceInfo.name} detected`)
+	}
+
+	static async create(ipcWrapper, devicepath) {
+		const shuttle = shuttleControlUSB
+		// We're doing device search via Companion so don't run it here too
+		shuttle.start(false)
+		try {
+			let deviceInfo = null
+			let info = null
+			shuttle.connect(devicepath)
+			deviceInfo = shuttle.getDeviceByPath(devicepath)
+			switch (deviceInfo.name) {
+				case 'ShuttleXpress':
+					info = shuttleXpressInfo
+					break
+				case 'ShuttlePro v1':
+					info = shuttleProV1Info
+					break
+				case 'ShuttlePro v2':
+					info = shuttleProV2Info
+					break
+				default:
+					this.ipcWrapper.log('debug', `Unknown Contour Shuttle device detected: ${this.deviceInfo.name}`)
+			}
+			if (!info) {
+				throw new Error('Unsupported model')
+			}
+
+			const self = new SurfaceUSBShuttle(ipcWrapper, devicepath, shuttle, info, deviceInfo)
+
+			await self.#init()
+
+			return self
+		} catch (e) {
+			shuttle.stop()
+
+			throw e
+		}
+	}
+
+	setConfig(config, force) {
+		// No config currently present
+		this.config = config
+	}
+
+	quit() {
+		try {
+			this.clearDeck()
+		} catch (e) {}
+
```

Comment:
Should I just drop this then given clearDeck doesn't exist/do anything for this device?
```suggestion
```

Reply:
yep, you can drop that